compiler-core/src/javascript/pattern.rs

```diff
@@ -346,22 +374,58 @@ impl<'module_ctx, 'expression_gen, 'a> Generator<'module_ctx, 'expression_gen, '
             Pattern::BitString { segments, .. } => {
                 use BitStringSegmentOption as Opt;
 
-                self.push_bitstring_length_check(subject.clone(), segments.len(), false);
-                self.push_string("buffer");
-                for (index, segment) in segments.iter().enumerate() {
+                let mut offset = Offset::new();
+                for segment in segments {
                     let _ = match segment.options.as_slice() {
-                        [] | [Opt::Int { .. }] => Ok(()),
+                        [] | [Opt::Int { .. }] => {
+                            self.push_method(format!("byteAt({})", offset.bytes));
+                            self.traverse_pattern(subject, &segment.value)?;
+                            self.pop();
+                            offset.increment(1);
+                            Ok(())
+                        }
+
+                        [Opt::Size { value: size, .. }] => match &**size {
+                            Pattern::Int { value, .. } => {
+                                let start = offset.bytes;
+                                offset.increment(value.parse::<usize>().unwrap() / 8);
+                                let end = offset.bytes;
+
+                                self.push_method(format!("intFromSlice({}, {})", start, end));
```

Comment:
this needs to be a custom method to handle transforming the slice to an integer. originally I tried just pushing values to the path. i.e. instead of `path.intFromSlice(1,2)` you'd have `path.buffer.slice(1,2)` which wouldn't require any additions to prelude but the return value from slice is still a utf8 array and not yet an int.

Reply:
Makes sense. Could we have a new Index variant per method? Better than passing strings around I think. Might want to rename "Index" to something else too...

Reply:
I don't quite understand what you mean here. do you mean `self.push_intFromSlice(start, end)`? is that for performance?

Reply:
I don't want to be Passing strings around when we could use a proper type. You get more help from the compiler if not using strings and it is the pattern that is already being used for the other checks.